### PR TITLE
Remove warnings/explicit settings on feature, restoration

### PR DESCRIPTION
**Pull request by alexdesiqueira**
## Description

Solves #4969.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by rfezzani — APPROVED** [review]
Thank you @alexdesiqueira :wink:

**Review by jni — APPROVED** [review]
Awesome, thanks @alexdesiqueira! Could you add these to the API changes of the dev release notes?

We should really implement towncrier...

File: skimage/feature/corner.py [hunk 1]
```diff
@@ -1052,18 +1050,12 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=None,
...
 
     """
     if threshold_rel is None:
         threshold_rel = 0.1
-        warn("Until version 0.16, threshold_rel was set to 0.1 by default. "
-             "Starting from version 0.16, the default value is set to None. "
-             "Until version 0.18, a None value corresponds to a threshold "
-             "value of 0.1. The default behavior will match "
-             "skimage.feature.peak_local_max. To avoid this warning, set "
-             "threshold_rel=0.", category=FutureWarning, stacklevel=2)
```
Comment (jni) [inline skimage/feature/corner.py:1066]: My reading of this is actually that the new default should be 0? Then *that* is what must go in the API change... Right?

File: skimage/restoration/_denoise.py [hunk 2]
```diff
@@ -815,14 +815,6 @@ def denoise_wavelet(image, sigma=None, wavelet='db1', mode='soft',
...
 
     if rescale_sigma is None:
-        msg = (
-            "As of scikit-image 0.16, automated rescaling of sigma to match "
-            "any internal rescaling of the image is performed. Setting "
-            "rescale_sigma to False, will disable this new behaviour. To "
-            "avoid this warning the user should explicitly set rescale_sigma "
-            "to True or False."
-        )
-        warn(msg, FutureWarning, stacklevel=2)
```
Comment (jni) [inline skimage/restoration/_denoise.py:825]: Also here: change the default value to `rescale=True` in the function signature.

**Review by jni — CHANGES_REQUESTED** [review]
@alexdesiqueira I misread this PR in my earlier review. If I'm not mistaken, `threshold_rel=0` should be the new default value for all the feature.functions, and `rescale_sigma=True` should go directly in the function signature for cycle spin.

**Comment by alexdesiqueira** [comment]
I'd ask for @sciunto's input as well; maybe I missed something, but #4969 reads different to me.

File: skimage/feature/corner.py (hunk 1 above)
Comment (sciunto) [reply skimage/feature/corner.py:1066]: In the signature, threshold_rel must be None. No modification of this variable must be done in this function. Therefore, the if statement has to be removed. This is to match feature.peak_local_max.

**Comment by sciunto** [comment]
In addition, please remove the corresponding lines in TODO.txt :)

**Comment by alexdesiqueira** [comment]
Alright @jni @sciunto that should suffice.
Please let me know what to do next. Maybe state that `rescale_sigma=True` on API changes?

File: doc/release/release_dev.rst [hunk 3]
```diff
@@ -38,6 +38,17 @@ API Changes
 - ``skimage.restoration.richardson_lucy`` returns a single-precision output
   when the input is single-precision. Prior to this release, double-precision
   was always used.
+- The explicit setting ``threshold_rel=0` was removed from the Examples of the
+  following docstrings: ``skimage.feature.BRIEF``,
+  ``skimage.feature.corner_harris``, ``skimage.feature.corner_shi_tomasi``,
+  ``skimage.feature.corner_foerstner``, ``skimage.feature.corner_fast``,
+  ``skimage.feature.corner_subpix``, ``skimage.feature.corner_peaks``,
+  ``skimage.feature.corner_orientations``, and
+  ``skimage.feature._detect_octave``.
```
Comment (jni) [inline doc/release/release_dev.rst:47]: This is not an API change.

File: skimage/feature/corner.py (hunk 1 above)
Comment (jni) [reply skimage/feature/corner.py:1066]: So, then, in API changes, we should list, "The default value of threshold_rel in feature.corner has changed from 0.1 to None, which corresponds to letting feature.peak_local_max decide on the default. That is currently equivalent to threshold_rel=0."

Something like that? The point is that the same function call (`corner_peaks(arr)`) will return different things between 0.17 and 0.18. That is what absolutely needs documenting in the release notes.

File: doc/release/release_dev.rst [hunk 4]
```diff
@@ -38,6 +38,17 @@ API Changes
...
+- In ``skimage.restoration._denoise``, the warning regarding
+  ``rescale_sigma=None`` was removed.
+- In ``skimage.restoration._cycle_spin``, the ``# doctest: +SKIP`` was removed.
```
Comment (jni) [inline doc/release/release_dev.rst:50]: Nor are these two. :joy:. See my comment below. Hopefully it is clear that only one API change has happened here!

**Review by jni — CHANGES_REQUESTED** [review]
@alexdesiqueira one tiny fix! Also, my suggestion about adding `.. versionchanged::`?

**Comment by jni** [comment]
> Also, my suggestion about adding `.. versionchanged::`?

LOL, I just remembered that was in a different PR :joy: Should apply here also though? :grimacing:

**Comment by alexdesiqueira** [comment]
> Nor are these two. joy. See my comment below. Hopefully it is clear that only one API change has happened here!

@jni hope this is it! Please check again :blush:

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
Boom. 😃

**Pull request merged**